lib/src/components/mixins/hitbox.dart

```diff
@@ -1,84 +1,70 @@
-  /// The hitbox is defined from the center of the component and with
-  /// percentages of the size of the component.
-  /// Example: [[1.0, 0.0], [0.0, 1.0], [-1.0, 0.0], [0.0, -1.0]]
-  /// This will form a square with a 45 degree angle (pi/4 rad) within the
-  /// bounding size box.
-  set shape(List<Vector2> vertices) => _shape = vertices;
-  List<Vector2> get shape => _shape ?? [];
+  UnmodifiableListView<HitboxShape> get shapes => UnmodifiableListView(_shapes);
 
-  /// Whether the hitbox shape has defined vertices and is not an empty list
-  bool hasShape() => _shape?.isNotEmpty ?? false;
+  void addShape(HitboxShape shape) {
+    shape.component = this;
+    _shapes.add(shape);
+  }
 
-  Iterable<Vector2> _scaledShape;
-  Vector2 _lastScaledSize;
+  void removeShape(HitboxShape shape) {
+    _shapes.remove(shape);
+  }
 
-  /// Gives back the shape vectors multiplied by the size of the component
-  Iterable<Vector2> get scaledShape {
-    if (_lastScaledSize != size || _scaledShape == null) {
-      _lastScaledSize = size;
-      _scaledShape = _shape?.map(
-        (p) => p.clone()..multiply(size / 2),
-      );
-    }
-    return _scaledShape;
+  /// Checks whether the hitbox represented by the list of [HitboxShape]
+  /// contains the [point].
+  @override
+  bool containsPoint(Vector2 point) {
+    return possiblyContainsPoint(point) &&
+        _shapes.any((shape) => shape.containsPoint(point));
   }
 
-  void renderContour(Canvas canvas) {
-    final hitboxPath = Path()
-      ..addPolygon(
-        scaledShape.map((point) => (point + size / 2).toOffset()).toList(),
-        true,
-      );
-    canvas.drawPath(hitboxPath, debugPaint);
+  void renderShapes(Canvas canvas) {
+    _shapes.forEach((shape) => shape.render(canvas, debugPaint));
   }
 
-  // These variables are used to see whether the bounding vertices cache is
-  // valid or not
-  Vector2 _lastCachePosition;
-  Vector2 _lastCacheSize;
-  double _lastCacheAngle;
-  bool _hadShape = false;
-  List<Vector2> _cachedHitbox;
+  /// Returns the absolute [Rect] that contains all the corners of the rotated
+  /// [toAbsoluteRect] rect.
+  Rect toBoundingRect() {
+    final rotatedPoints = toAbsoluteRect().toVectors()
+      ..forEach((v) => v.rotate(
+            angle,
+            center: absolutePosition,
+          ));
+    final minX =
+        rotatedPoints.map<double>((v) => v.x).fold(double.infinity, min);
+    final minY =
+        rotatedPoints.map<double>((v) => v.y).fold(double.infinity, min);
+    final maxX = rotatedPoints
+        .map<double>((v) => v.x)
+        .fold(double.negativeInfinity, max);
+    final maxY = rotatedPoints
+        .map<double>((v) => v.y)
+        .fold(double.negativeInfinity, max);
 
-  bool _isHitboxCacheValid() {
-    return _lastCacheAngle == angle &&
-        _lastCacheSize == size &&
-        _lastCachePosition == position &&
-        _hadShape == hasShape();
+    return Rect.fromPoints(Offset(minX, minY), Offset(maxX, maxY));
```

Comment:
how about `Vector2(minX, minY) & Vector2(maxX, maxY)`? same but prettier :P

Reply:
You mean to add another helper on the `Rect` extension or to create `Vector2` and then convert them to `Offset`?
(This is the dart Rect, not my Rectangle)

Reply:
we already have an & operator on Vector2 that creates dart Rects

Reply:
I think @wolfenrain added it

Reply:
I did... Did not mean to commit that but I did :D

Reply:
It is not merged yet tho @luanpotter

Reply:
ah! let's leave it though, we can refactor after both PRs are in :)